datafusion/core/src/lib.rs

```diff
@@ -678,6 +678,12 @@ doc_comment::doctest!(
     library_user_guide_sql_api
 );
 
+#[cfg(doctest)]
```

Comment:
I added this block which means the examples in the library guide are now run as part of `cargo test --doc`